.github/workflows/linting-and-tests.yml

```diff
@@ -378,7 +378,7 @@ jobs:
         uses: docker/setup-buildx-action@v2
 
       - name: Build engine Docker image locally # using github actions docker cache
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v4
```

Comment:
this addresses all of these deprecation warnings: 
<img width="1120" alt="Screenshot 2023-08-04 at 11 52 27" src="https://github.com/grafana/oncall/assets/9406895/3f85e07c-8a20-4f2c-ad49-6b8d40345819">